dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaEventListenerQaService.kt

```diff
@@ -215,8 +215,27 @@ class QaEventListenerQaService
             MessageQueueUtils.rejectMessageOnException {
                 val dataId = MessageQueueUtils.readMessagePayload<DataUploadedPayload>(payload, objectMapper).dataId
                 MessageQueueUtils.validateDataId(dataId)
+
+                val qaReviewEntity = qaReviewManager.getMostRecentQaReviewEntity(dataId)!!
+                val activeDataId =
+                    qaReviewManager.getDataIdOfCurrentlyActiveDataset(
+                        qaReviewEntity.companyId, qaReviewEntity.framework, qaReviewEntity.reportingPeriod,
+                    )
+                val deletedDataIdWasActive = (dataId == activeDataId)
+
                 qaReportManager.deleteAllQaReportsForDataId(dataId, correlationId)
                 qaReviewManager.deleteAllByDataId(dataId, correlationId)
+                if (deletedDataIdWasActive) {
```

Comment:
```suggestion
                if (deletedDataId == activeDataId ) {
```

und im if statement auf newActiveDataId not null überprüfen . Dann kann man das .toString auch weglassen